src/clue/adafruit_slideshow.py

```diff
@@ -0,0 +1,312 @@
+from PIL import Image
+
+import os
+import base64
+from io import BytesIO
+from base_circuitpython import base_cp_constants as CONSTANTS
+import time
+import collections
+from random import shuffle
+from common import utils
+
+# taken from adafruit
+# https://github.com/adafruit/Adafruit_CircuitPython_Slideshow/blob/master/adafruit_slideshow.py
+
+
+class PlayBackOrder:
+    """Defines possible slideshow playback orders."""
+
+    # pylint: disable=too-few-public-methods
+    ALPHABETICAL = 0
+    """Orders by alphabetical sort of filenames"""
+
+    RANDOM = 1
+    """Randomly shuffles the images"""
+    # pylint: enable=too-few-public-methods
+
+
+class PlayBackDirection:
+    """Defines possible slideshow playback directions."""
+
+    # pylint: disable=too-few-public-methods
+    BACKWARD = -1
+    """The next image is before the current image. When alphabetically sorted, this is towards A."""
+
+    FORWARD = 1
+    """The next image is after the current image. When alphabetically sorted, this is towards Z."""
+    # pylint: enable=too-few-public-methods
+
+
+# custom
+class SlideShow:
+    def __init__(
+        self,
+        display,
+        backlight_pwm=None,
+        *,
+        folder=".",
+        order=PlayBackOrder.ALPHABETICAL,
+        loop=True,
+        dwell=3,
+        fade_effect=True,
+        auto_advance=True,
+        direction=PlayBackDirection.FORWARD,
+    ):
+        self._BASE_DWELL = 0.3
+        self._BASE_DWELL_DARK = 0.7
+        self._NO_FADE_TRANSITION_INCREMENTS = 18
+
+        self.auto_advance = auto_advance
+        """Enable auto-advance based on dwell time.  Set to ``False`` to manually control."""
+
+        self.loop = loop
+        """Specifies whether to loop through the images continuously or play through the list once.
+        ``True`` will continue to loop, ``False`` will play only once."""
+
+        self.fade_frames = 8
+        """Whether to include the fade effect between images. ``True`` tells the code to fade the
+           backlight up and down between image display transitions. ``False`` maintains max
+           brightness on the backlight between image transitions."""

```

Comment:
Kind of confused with the value and the description. Value is 8 but description makes it seem like it is just a boolean.

Reply:
oops, I put the doctstring in the wrong place!